runtime/doc/eval.txt

```diff
@@ -3507,9 +3507,22 @@ getcurpos()	Get the position of the cursor.  This is like getpos('.'), but
 			MoveTheCursorAround
 			call setpos('.', save_cursor)
 <
-							*getcwd()*
-getcwd()	The result is a String, which is the name of the current
-		working directory.
+getcwd([{window}[, {tab}]])					*getcwd()*
+		With no argument the result is a String, which is the name of the
+		current effective working directory. If an arguments are provided for a
+		window or a tabpage the working directory of that scope is returned.
+
+		Tabs and windows are identified using their respective number. If a
+		number is 0 the current tab or window is implied. If one is not given
+		it defaults to 0. Thus
+		
+			getcwd()
+			getcwd(0)
+			getcwd(0, 0)
+
+		are all equal. If {window} is -1 it will be ignored and only the tab
+		will be looked at.
```

Comment:
Local working directories cascade upwards: The lowest one is returned, unless the arguments is -1. If there is not a lowest one the next higher one is returned. To answer your question, the code tries to find the local working directory of the window, if there isn't one it cascades to the tab-local, and if there isn't one either it falls back to the global working directory.